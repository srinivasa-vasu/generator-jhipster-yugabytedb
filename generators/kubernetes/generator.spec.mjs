import prom from 'expect';

import { helpers, lookups } from '#test-utils';

const SUB_GENERATOR = 'kubernetes';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;
const { expect } = prom;

describe('SubGenerator kubernetes of yugabytedb JHipster blueprint', () => {
  describe('micro', () => {
    let result;
    before(async function () {
      result = await helpers
        .create(BLUEPRINT_NAMESPACE)
        .withOptions({
          reproducible: true,
          defaults: true,
          blueprint: 'yugabytedb',
        })
        .withPrompts({
          deploymentApplicationType: 'microservice',
          directoryPath: './',
          adminPassword: 'meetup',
          dockerRepositoryName: 'jhipsterrepository',
          dockerPushCommand: 'docker push',
          kubernetesNamespace: 'jhipsternamespace',
          istio: false,
        })
        .withLookups(lookups)
        .run();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe('mono', () => {
    let result;
    before(async function () {
      result = await helpers
        .create(BLUEPRINT_NAMESPACE)
        .withOptions({
          reproducible: true,
          defaults: true,
          blueprint: 'yugabytedb',
        })
        .withPrompts({
          deploymentApplicationType: 'monolith',
          directoryPath: './',
          adminPassword: 'meetup',
          dockerRepositoryName: 'jhipsterrepository',
          dockerPushCommand: 'docker push',
          kubernetesNamespace: 'jhipsternamespace',
          istio: false,
        })
        .withLookups(lookups)
        .run();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
