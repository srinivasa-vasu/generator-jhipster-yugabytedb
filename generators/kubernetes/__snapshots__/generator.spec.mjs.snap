// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubGenerator kubernetes of yugabytedb JHipster blueprint micro should succeed 1`] = `
Object {
  ".yo-rc.json": Object {
    "stateCleared": "modified",
  },
  "kubectl-apply.sh": Object {
    "stateCleared": "modified",
  },
  "kustomization.yml": Object {
    "stateCleared": "modified",
  },
  "namespace.yml": Object {
    "stateCleared": "modified",
  },
  "skaffold.yml": Object {
    "stateCleared": "modified",
  },
}
`;

exports[`SubGenerator kubernetes of yugabytedb JHipster blueprint mono should succeed 1`] = `
Object {
  ".yo-rc.json": Object {
    "stateCleared": "modified",
  },
  "kubectl-apply.sh": Object {
    "stateCleared": "modified",
  },
  "kustomization.yml": Object {
    "stateCleared": "modified",
  },
  "namespace.yml": Object {
    "stateCleared": "modified",
  },
  "skaffold.yml": Object {
    "stateCleared": "modified",
  },
}
`;
